src/Form/Formbot.js

```diff
@@ -137,8 +148,12 @@ export default class Formbot extends React.Component {
       let errorMsg;
 
       try {
-        if (fromSchema) {
-          validation.validateSync(fieldValue, { values: this.state.values });
+        if (hasSchema && typeof validation.validate === 'function') {
+          validation.validate(fieldValue, validationOpts).catch(e => {
+            this.setErrors({ [field]: e.message }, resolve);
```

Comment:
Why is it necessary to catch the error here and not let it be caught and handled in the catch/finally blocks below?

Reply:
@schlegz It's a promise and the error is generated not when the `try/catch` block is running but later so there's no way to catch that promise rejection with a `try/catch`. 


```js
const error = async () => throw new Error();

const validate = () => new Promise((resolve) => {
  try {
    error();
    resolve();
  } catch (e) { 
    //  promise rejection will not be caught here
  }
})
```